django/db/backends/mysql/base.py

```diff
@@ -308,8 +308,9 @@ def check_constraints(self, table_names=None):
                 primary_key_column_name = self.introspection.get_primary_key_column(cursor, table_name)
                 if not primary_key_column_name:
                     continue
-                key_columns = self.introspection.get_key_columns(cursor, table_name)
-                for column_name, referenced_table_name, referenced_column_name in key_columns:
+                relations = self.introspection.get_relations(cursor, table_name)
```

Comment:
Did you consider?
`for column_name, (referenced_column_name, referenced_table_name) in relations.items():`
(and similar for sqlite3)